notebook/notebookapp.py

```diff
@@ -1566,15 +1564,23 @@ def init_server_extensions(self):
         manager = ConfigManager(read_config_path=config_path)
         section = manager.get(self.config_file_name)
         extensions = section.get('NotebookApp', {}).get('nbserver_extensions', {})
+        
+        for modulename, enabled in sorted(extensions.items()):
+            if modulename not in self.nbserver_extensions:
+                self.config.NotebookApp.nbserver_extensions.update({modulename: enabled})
```

Comment:
This makes it very likely that `self.config.NotebookApp.nbserver_extensions` will be a LazyConfigValue, which is not meant to be interrogated directly, so you will still have trouble if you ever try to talk to the config object directly.